### PR TITLE
4.5.9 plat 7163 distinguish old recording to new

### PR DESCRIPTION
**Pull request by gadyaari**

### CODE REVIEW

File: KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/RecordingModule.java [hunk 1]
```diff
@@ -143,11 +143,12 @@ public void run() {
                         setGroupPermision(filePath);
                     }
 
-                     updatedEntry = appendRecording(liveEntry, liveAsset.id, index, filePath, appendTime, lastChunkFlag);
-                    if (updatedEntry != null){
-                        liveEntry = updatedEntry;
+                    if (!KalturaAPI.getKalturaAPI().isNewRecordingEnabled(liveEntry)) {
```
Comment (kalturaguy) [inline KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/RecordingModule.java]: it's bad practive todo API call over and over when you already know the answer...

File: KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/RecordingModule.java [hunk 2]
```diff
@@ -164,7 +167,9 @@ public void onUnPublish() {
             if (liveAsset.tags.contains(Constants.RECORDING_ANCHOR_TAG_VALUE) && KalturaEntryServerNodeType.LIVE_PRIMARY.equals(index)) {
                 logger.info("Cancel replacement is required");
                 if (liveEntry.recordedEntryId != null && liveEntry.recordedEntryId.length() > 0) {
-                    KalturaAPI.getKalturaAPI().cancelReplace(liveEntry);
+                    if (!isNewLiveRecordingEnabled) {
```
Comment (kalturaguy) [inline KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/RecordingModule.java]: please fix if in if in if

File: KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/RecordingModule.java [hunk 3]
```diff
@@ -84,6 +85,7 @@ public FlavorRecorder(KalturaLiveEntry liveEntry, KalturaLiveAsset liveAsset, Ka
             this.index = index;
             this.isLastChunk = false;
 
+            this.isNewLiveRecordingEnabled = KalturaAPI.getKalturaAPI().isNewRecordingEnabled(liveEntry);
```
Comment (kalturaguy) [inline KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/RecordingModule.java]: better to do it once instead of per flavor

Comment (gadyaari) [inline KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/RecordingModule.java]: Can't avoid calling each flavor for the permission. There's no recording entity per entry

**Pull request merged**